src/librbd/io/ImageRequestWQ.cc

```diff
@@ -525,6 +537,23 @@ void ImageRequestWQ<I>::set_require_lock(Direction direction, bool enabled) {
 }
 
 template <typename I>
+void ImageRequestWQ<I>::apply_qos_iops_limit(uint64_t limit) {
+  iops_throttle->set_max(limit);
+  iops_throttle->set_average(limit);
+}
+
+template <typename I>
+void ImageRequestWQ<I>::handle(int r, ImageRequest<I> *item) {
+  assert(m_io_blockers.load() > 0);
+  --m_io_blockers;
+  if (r == 0) {
```

Comment:
How would it fail?

Reply:
Never, at least currently.  So remove it, and add it when it would really happen.